plenum/common/ledger_manager.py

```diff
@@ -875,6 +882,38 @@ def mark_ledger_synced(self, ledger_id):
                    for l in self.ledgerRegistry.values()):
                 self.postAllLedgersCaughtUp()
 
+    def catchup_next_ledger(self, ledger_id):
+        if self.ownedByNode:
+            next_ledger_id = self.ledger_to_sync_after(ledger_id)
+            if next_ledger_id:
+                self.catchup_ledger(next_ledger_id)
+            else:
+                logger.debug('{} not found any ledger to catchup after {}'
+                             .format(self, ledger_id))
+        else:
+            logger.debug('{} not owned by node'.format(self))
+
+    def catchup_ledger(self, ledger_id):
+        try:
+            ledger_info = self.getLedgerInfoByType(ledger_id)
+            ledger_info.set_defaults()
+            ledger_info.canSync = True
+            self.owner.request_ledger_status_from_nodes(ledger_id)
+            self.processStashedLedgerStatuses(ledger_id)
+        except KeyError:
+            logger.error("ledger type {} not present in ledgers so "
+                         "cannot set state".format(ledger_id))
+            return
+
+    def ledger_to_sync_after(self, ledger_id) -> Optional[int]:
+        if self.ledger_sync_order:
```

Comment:
I am not seeing the difference it makes to clarity in this case as it is just one condition, its like saying if condition is true then do this vs if condition is false stop else move ahead

Reply:
I agree that this is not critical when there is just one condition. But I think the code like
```
if condition:
    <10 lines of code>
```
is not very readable because of unnecessary tab.